### DISCUSSION TITLE
String to ebiten.Key

### DISCUSSION THREAD

**Discussion by mgord9518** [hajimehoshi/ebiten#2132] [Q&A]
I'm trying to save custom control layouts to a config file, but there seems to be no easy way to convert a string value to ebiten.Key in the API reference so I have no clue how to actually set the keys in game after loading my config.

The opposite is incredibly easy with ebiten.Key.String() and I see keyNameToKeyCode() in the ebiten source, but it's unexported so I can't use it without just duplicating the entire function

**Comment by hajimehoshi** [comment]
Hi,

The problem of converting a string to a key is that it is not obvious how to treat multiple possibility: For `KeyEnter`, should we accept both `enter` and `return`? What about `KeyApostrophe` (`apostrophe` vs `quote` vs `'`)?

**Reply by mgord9518** [reply]
Why not just only accept a single value like the ones returned from ebiten.Key.String()? Eg: ebiten.KeyEnter would only be returned from the string "Enter" (non case sensitive) while things like "Return" would return an error

Then a list of valid key strings could be linked in the API guide

**Reply by hajimehoshi** [reply]
https://github.com/hajimehoshi/ebiten/issues/2133 I created an issue for this. Apparently there is a way to handle by yourself. Please take a look at the comment. I'm still thinking whether to introduce a new API.

**Reply by mgord9518** [reply]
I just ended up just using the map from their suggestion which works with my needs fairly elegantly, but it does feel like something useful enough to warrant adding an API

**Reply by hajimehoshi** [reply]
OK I've implemented `(*Key).UnmarshalText`.
